file-attribute-caching/src/test/kotlin/com/pkware/filesystem/attributecaching/AttributeCachingFileSystemTests.kt

```diff
@@ -31,10 +33,52 @@ import java.nio.file.attribute.UserPrincipal
 import java.text.SimpleDateFormat
 import java.util.EnumSet
 import java.util.stream.Stream
+import kotlin.concurrent.thread
 import kotlin.io.path.exists
 
 class AttributeCachingFileSystemTests {
 
+    @Disabled // Enable this test if you wish to see how the attribute caching fileSystem fails in a multi-threaded
+    // environment. This test will be very inconsistent, but it will throw a ClassCastException when it does fail
+    // (you may have to run it multiple times).
+    // The ClassCastException occurs within the map used for the FileSystems stored within the
+    // AttributeCachingFileSystemProvider.
+    @Test
+    fun `attribute caching filesystem invoked many times across multiple threads throws ClassCastException`() {
+        var caughtException: Exception? = null
+        val defaultFileSystem = FileSystems.getDefault()
+        val loopCount = 100_000
+        val waitTimeMillis: Long = 50_000
+
+        fun runTestThread(): Thread {
+            return thread {
+                for (i in 1..loopCount) {
+                    try {
+                        AttributeCachingFileSystem.wrapping(defaultFileSystem).use { }
+                    } catch (e: Exception) {
+                        // Test fail fast and verify the exception thrown here
+                        assertThat(e).isInstanceOf(ClassCastException::class.java)
+                        caughtException = e
+                    }
```

Comment:
```suggestion
                    } catch (e: ClassCastException) {
                        caughtException = e
                    }
```

Reply:
This would also give visibility into a different exception being thrown, but be less verbose

Reply:
this test never threw a ConcurrentModificationException and we never came up with a test for that- so we shouldnt be worried about that here

this test was to demonstrate first that the filesystem creation wasn't thread safe, but then you wanted specifics as far as the exception that was generated so I have that here now

The test looking for a specific exception is fine as is.

I will instead make 2 separate commits, one with this test as it is and one where it(s) listed as not throwing anything because the fix has been applied and it can always be enabled

Reply:
I didn't ever want specifics on the exception to my recollection. I wanted visibility only because it would sometimes fail silently. 

I agree it's fine as is for performing in the task, only providing commentary that it is a very strange inclusion, especially since we no longer expect this phenomenon, given the introduced fix.

Reply:
It would make more sense if this were a test that did this kind of thing and asserted its expectation to _not_ produce an exception, so that when the problem is fixed (which we believe it is), we enable it so we don't regress in this particular way.

The reason I'm providing commentary on this at all is only because even with that in mind, the test is still weird. Yes, I know I wrote it drafts of it lol. But I didn't write them with the intent for them to be included in the repo verbatim, only to provide a reproduce of an issue.

Reply:
i ended up going this route, I just wanted a failing test first but at this point I am just getting this in to be done